File: app/src/main/java/com/example/materialepladsenappui/Database_Connection/DBHelper.java
package com.example.materialepladsenappui.Database_Connection;

import static java.sql.DriverManager.println;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class DBHelper {
    public static String selectSome() throws SQLException {
        String response = "";

        DatabaseConnection db = new DatabaseConnection();
        Connection conn = db.getConnection();

        String query = "SELECT TOP (1000) [product_group]\n" +
                "      ,[id]\n" +
                "      ,[product_articlenumber]\n" +
                "      ,[product_articleName]\n" +
                "      ,[product_size]\n" +
                "      ,[product_WeightM3]\n" +
                "      ,[product_price_dkk_excludingTax]\n" +
                "      ,[product_price_dkk_includingTax]\n" +
                "      ,[unit]\n" +
                "      ,[availability]\n" +
                "      ,[product_image]\n" +
                "  FROM [materialepladsen_core_DTUMP2].[dbo].[v_mobileApp_products]";

        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(query);

        while (rs.next()) {
            String tableName = rs.getString("TABLE_NAME");
            println("Table name: $tableName");
        }

        return response;
    }
}

File: app/src/main/java/com/example/materialepladsenappui/Database_Connection/DBHelper.kt
package com.example.materialepladsenappui.Database_Connection

import android.content.ContentValues.TAG
import android.nfc.Tag
import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.sql.Connection

class DBHelper {
    // Method to select a record from the database
    fun Select() {
        // Create a DatabaseConnection object and get a connection
        val db = DatabaseConnection()
        val conn = db.getConnection()

        // Check if the connection is not null and not closed
        if (conn != null) {
            if (conn.isClosed) {
                Log.i("TestConn", "Connection is closed")
            } else {
                Log.i("TestConn", "Connection is not closed " + conn.toString())
            }
        } else {
            Log.i("TestConn", "Connection is null")
        }

        // Set the query to select a record from the v_mobileApp_products table
        val query = "SELECT [product_group] FROM [materialepladsen_core_DTUMP2].[dbo].[v_mobileApp_products]"

        // Create a statement and execute the query
        val stmt = conn?.createStatement()
        Log.i("TestConn", "after statement")
        val rs = stmt?.executeQuery(query)
        Log.i("TestConn", "after executeQuery")

        // Log the result set
        Log.i("TestConn", rs.toString())
        while (rs?.next() == true){
            Log.i("TestConn",rs.getString(1))
        }
    }
}